src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -310,8 +308,8 @@ typedef struct ucc_tl_mlx5_mcast_coll_comm {
     ucc_rank_t                                      commsize;
     char                                           *grh_buf;
     struct ibv_mr                                  *grh_mr;
-    uint16_t                                        mcast_lid;
-    union ibv_gid                                   mgid;
+    uint16_t                                       *lid_list;
```

Comment:
maybe move lid and gid to mcast_ctx and make array of structures instead? For example
```
struct mcast_group {
    struct ibv_qp *qp;
    struct ibv_ah *ah;
    uint16_t mlid;
    union ibv_gid mgid;
    struct ibv_send_wr  swr;
    struct ibv_sge      ssg;
}

struct mcast_ctx {
    struct ibv_qp      *qp;
    struct ibv_ah      *ah;
    struct ibv_send_wr  swr;
    struct ibv_sge      ssg;

    // RC connection info for supporing one-sided based relibality
    struct ibv_qp     **rc_qp;
    uint16_t           *rc_lid;
    union ibv_gid      *rc_gid;

    // multiple mcast group
    struct mcast_group_t *mcast_groups;
};

```

Reply:
Good point! I made all these changes. please see the new changes